File: harddrive.py
from perangkat import Perangkat

class Harddrive(Perangkat):
    def __init__(self, merk, nama, kapasitas_gb, tipe_drive):
        super().__init__(merk, nama)
        self.kapasitas_gb = kapasitas_gb
        self.tipe_drive = tipe_drive
    
    def info(self):
        return f"Harddrive: {self.tipe_drive} {self.merk} {self.nama} ({self.kapasitas_gb}GB)"

File: harddrive.cpp
#include "harddrive.h"

Harddrive::Harddrive(std::string merk, std::string nama, int kapasitas_gb, std::string tipe_drive)
    : Perangkat(merk, nama), kapasitas_gb(kapasitas_gb), tipe_drive(tipe_drive) {}

std::string Harddrive::info() const {
    return "Harddrive: " + tipe_drive + " " + merk + " " + nama + " (" + std::to_string(kapasitas_gb) + "GB)";
}
